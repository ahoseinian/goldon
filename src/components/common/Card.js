import styled from 'styled-components'

const Card = styled.div `
  border-radius: 0 !important;
`
export default Card
