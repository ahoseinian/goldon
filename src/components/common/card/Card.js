import styled from 'styled-components'
import {color} from '../../theme'

const Card = styled.div `
  border-radius: 0 !important;
  box-shadow: 0 7px 18px -7px ${color.dark};
`
export default Card
