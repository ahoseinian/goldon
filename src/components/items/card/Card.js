import React from 'react'
import styled from 'styled-components'
import {connect} from 'react-redux'
import ItemCard from '../../common/card/Card'
import {addItemToCart} from '../../../actions'
import ItemImage from '../image/Image'
import Sizes from './Sizes'
import PriceRow from './PriceRow'


const ItemName = styled.div `
  text-align:center;
  font-size: 1.2rem;
  line-height: 4rem;
  max-height: 4rem;
`

export const Card = ({item, addItemToCart}) => (
  <ItemCard className="card">
    <ItemName>{item.name}</ItemName>
    <ItemImage id={item.images[0]}/>
    <PriceRow price={item.price} handleClick={addItemToCart}/>
    <Sizes size={item.size}/>
  </ItemCard>
)

Card.propTypes = {
  item: React.PropTypes.shape({
    name: React.PropTypes.string,
    images: React.PropTypes.array,
    price: React.PropTypes.number,
    size: React.PropTypes.object
  }),
  addItemToCart: React.PropTypes.func
}

export default connect(null, ({addItemToCart}))(Card)
