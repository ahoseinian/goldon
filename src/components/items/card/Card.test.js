import React from 'react'
import {shallow} from 'enzyme'
import {Card} from './Card'
import toJson from 'enzyme-to-json'

const item = {
  name: '',
  images: ['one'],
  price: 22,
  size: {},
}

describe('Card', () => {
  it('renders', () => {
    const wrapper = shallow(<Card item={item} addItemToCart={jest.fn()}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
