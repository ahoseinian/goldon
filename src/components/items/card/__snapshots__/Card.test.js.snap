exports[`Card renders 1`] = `
<styled.div
  className="card">
  <styled.div />
  <ItemImage
    id="one" />
  <PriceRow
    handleClick={[Function]}
    price={22} />
  <Sizes
    size={Object {}} />
</styled.div>
`;
